pkg/kubelet/util/pluginwatcher/README

```diff
@@ -0,0 +1,29 @@
+This folder contains a utility, pluginwatcher, for Kubelet to register
+different types of node-level plugins such as device plugins or CSI plugins.
+It discovers plugins by monitoring inotify events under the directory returned by
+kubelet.getPluginsDir(). Lets refer this directory as PluginsSockDir.
+For any discovered plugin, pluginwatcher issues Registration.GetInfo grpc call
```

Comment:
Is it too late to turn watcher into a service? The plugin calls the watcher at a prescribed UDS location to register itself instead of the watcher doing inotify on a flat directory.  That way the watcher can apply stricter registration rules including voiding name clashes, etc.

Reply:
If i understood correctly, you are suggesting model 1, where plugin starts the communication. As explained the design doc, there is a problem in re-registration case with model 1. Therefore we decided to implement model 2. 
flat directory is just for initial implementation and will be extended to hierarchical in successive PR. 
I think in model 1 also onus of avoiding plugin name conflict is on plugins. I am not sure how model 1 is better than model 2 in avoiding name clashes.

Reply:
@vikaschoudhary16 I am concerned with plugin name clash between plugin types of CSI and DevicePlugin plugins.  If PluginsSockDir is flat then a plugin name from CSI may clash with a plugin from DevicePlugin.